spec/features/services/reverting_service_spec.rb

```diff
@@ -0,0 +1,153 @@
+require 'rails_helper'
+
+describe 'Reverting to previous services', js: true do
+  before :all do
+    # service for simple reverting service test
+    @simple_revert_ingest_response, @simple_revert_concept_response = publish_service_draft(revision_count: 2)
+
+    # service for reverting service with many revisions
+    @multiple_revisions_ingest_response, @multiple_revisions_concept_response = publish_service_draft(revision_count: 4, long_name: 'Reverting Services Test', number_revision_long_names: true)
+  end
+
+  before do
+    login
+  end
+
+  context 'when the latest revision is a published service' do
+    before do
+      visit service_path(@simple_revert_ingest_response['concept-id'])
+
+      click_on 'Revisions'
+    end
+
+    it 'displays the correct phrasing for reverting records' do
+      expect(page).to have_content('Revert to this Revision', count: 1)
+    end
+
+    context 'when reverting the service' do
+      before do
+        click_on 'Revert to this Revision'
+        click_on 'Yes'
+
+        wait_for_ajax
+        wait_for_cmr
+      end
+
+      it 'displays all the correct revision information' do
+        expect(page).to have_content('Revision Created Successfully!')
+
+        expect(page).to have_content('Published', count: 1)
+        expect(page).to have_content('Revision View', count: 2)
+        expect(page).to have_content('Revert to this Revision', count: 2)
+      end
+    end
+
+    context 'when reverting to a revision before the previous revision from a different provider context' do
+      context 'when visiting the revisions page from a different provider' do
+        before do
+          user = User.first
+          user.provider_id = 'MMT_1'
+          user.available_providers = %w(MMT_1 MMT_2)
+          user.save
+
+          visit service_revisions_path(@multiple_revisions_ingest_response['concept-id'])
+        end
+
+        it 'displays all the correct revision information' do
+          within 'main header' do
+            expect(page).to have_content('Reverting Services Test -- revision 04')
+          end
+
+          expect(page).to have_content('Published', count: 1)
+          expect(page).to have_content('Revision View', count: 3)
+          expect(page).to have_content('Revert to this Revision', count: 3)
+        end
+
+        context 'when reverting to the earliest revision' do
+          before do
+            visit service_revisions_path(@multiple_revisions_ingest_response['concept-id'])
+
+            within '#service-revisions-table tbody tr:last-child' do
+              # make sure we are clicking on the correct link
+              expect(page).to have_content('1 - Revision')
+
+              click_on 'Revert to this Revision'
+            end
+
+            find('.not-current-provider-link').click
```

Comment:
I see here that you're clicking on the not current provider modal, but there don't seem to be any tests for the modal and switching providers for reverting. Should we have that here?

Reply:
yeah, looks like it is missing